src/ogdf/fileformats/DotLexer.cpp

```diff
@@ -198,25 +198,34 @@ bool Lexer::tokenizeLine()
 }
 
 
-bool Lexer::match(const Token::Type &type)
+bool Lexer::match(const Token::Type &type, const bool word)
 {
-	return match(Token::toString(type));
+	return match(Token::toString(type), word);
 }
 
 
-bool Lexer::match(const std::string &str)
+bool Lexer::match(const std::string &str, const bool word)
 {
+	const size_t strLength = str.length();
 	// Check whether buffer is too short to match.
-	if(m_buffer.length() - m_col < str.length()) {
+	if(m_buffer.length() < m_col + strLength) {
 		return false;
 	}
 
-	for(size_t i = 0; i < str.length(); i++) {
+	for(size_t i = 0; i < strLength; i++) {
 		if(m_buffer[m_col + i] != str[i]) {
 			return false;
 		}
 	}
 
+	// we've matched a part of a word instead of a whole word
```

Comment:
I think the comment refers to the `if` on line 223

Reply:
Preferably, you can merge these `if`s to `if (word && m_buffer.length() >= m_col + str.length() + 1 && isalnum(m_buffer[m_col + str.length()]))`